presto-parquet/src/main/java/com/facebook/presto/parquet/reader/MetadataReader.java

```diff
@@ -124,11 +125,14 @@ public static ParquetMetadata readFooter(FSDataInputStream inputStream, Path fil
                             .map(value -> value.toLowerCase(Locale.ENGLISH))
                             .toArray(String[]::new);
                     ColumnPath columnPath = ColumnPath.get(path);
-                    PrimitiveTypeName primitiveTypeName = messageType.getType(columnPath.toArray()).asPrimitiveType().getPrimitiveTypeName();
+                    final org.apache.parquet.schema.Type type = messageType.getType(columnPath.toArray());
+                    final PrimitiveType primitiveType = messageType.getType(columnPath.toArray()).asPrimitiveType();
+                    PrimitiveTypeName primitiveTypeName = primitiveType.getPrimitiveTypeName();
                     ColumnChunkMetaData column = ColumnChunkMetaData.get(
                             columnPath,
-                            primitiveTypeName,
+                            primitiveType,
                             CompressionCodecName.fromParquet(metaData.codec),
+                            null, //TODO: encoding stats
```

Comment:
Can you please add more details to this `TODO` as it's not clear to me what needs to be done here?

Reply:
Removed the TODO and added the required stats.